Maximizing the consistency in the dataset documentation

koshtiakanksha opened the pull request:
fix #178

koshtiakanksha commented:
Hello @cregouby, I’m currently working on issue #178, just wanted to check in and confirm whether I’m heading in the right direction before proceeding to update the remaining files.

cregouby commented on R/dataset-eurosat.R:
```diff
@@ -15,7 +16,9 @@
 #' @param target_transform Function. Optional transformation to be applied to the labels.
 #'
-#' @return A `torch::dataset` object named x and y with x, a 64x64 image with 3 or 13 layers, and y, the label .
+#' @return An object of class `eurosat_dataset`. Each item is a list with:
```
**suggestion** I do prefer "A `torch::dataset` object." than "An object of class `eurosat_dataset`.". In the R end-user community, nobody cares (or even understand) about object class names. What they have to know is that it is a torch::dataset, so they can find documentation of what to apply later on...

koshtiakanksha commented:
Hello @cregouby 
- Done with #178 (consistent documentation): ensured consistent use of @inheritParams across all updated dataset files
- Verified output naming convention as x and y (#180 - output names)
- Incorporated standardized long-step user messaging for download & extraction (#179 - user messaging)
- Unified warning/info messages using consistent variable naming (#183 - user messages)
- Also addressed #182 (maximize reuse of parameter docs)
- Will recheck for consistency in upcoming pass

cregouby commented on R/dataset-cifar.R:
```diff
@@ -1,19 +1,25 @@
+#' item$x
+#' item$y
+#' }
+#' @family datasets
```
**improvement** Could we specialize a bit and move to `@family classification_dataset` ? This would be consistent with what we build in the pkgdown website.

cregouby commented on R/dataset-caltech.R:
**praise** I like it a lot.
**improvement** **blocking** Please have a look at the result in RStudio help pannel : You also have to generalize the description : 
![image](https://github.com/user-attachments/assets/fa5276dd-355f-48ea-a410-f222494009cd)
**improvement** and also make 2 paragraphs in the details 
![image](https://github.com/user-attachments/assets/6df0882d-0635-407e-a4ab-c744486394be)

cregouby reviewed (changes requested):
**praise** This is a strong improvement, thanks.
**improvement** I would split the @family into `classification_dataset`, ... (see comments)

cregouby commented on R/dataset-eurosat.R:
**todo** General description conatains duplicate information and shall be generalized to the 3 datasets
![image](https://github.com/user-attachments/assets/322995bc-ff54-4210-a305-209791b714bc)
**improvement** @rdname eurosat_dataset (singular) would save one redundant entry in the package function list
![image](https://github.com/user-attachments/assets/aa538661-1f09-4093-88cd-6d6363c5b0a1)

cregouby commented on R/dataset-fer.R:
```diff
@@ -72,17 +70,15 @@ fer_dataset <- dataset(
     }
 
-    cli::cli_inform("Parsing image data into 48x48 grayscale arrays...")
+    cli_inform("{.cls {class(self)[[1]]}} Parsing image data into 48x48 grayscale arrays...")
```
**todo** can we switch to the common message ? (remove the specifics.)

cregouby commented on R/dataset-fer.R:
```diff
@@ -72,17 +70,15 @@ fer_dataset <- dataset(
-    cli::cli_inform(
-      "FER-2013 ({self$split}) loaded: {length(self$x)} images, 48x48 grayscale, {length(self$classes)} classes."
-    )
+    cli_inform("{.cls {class(self)[[1]]}} Parsing image data into 48x48 grayscale arrays...")
```
**todo** idem

cregouby commented on R/dataset-fer.R:
```diff
@@ -107,21 +103,22 @@ fer_dataset <- dataset(
   download = function() {
     if (self$check_files()) {
-      cli::cli_inform("FER-2013 already exists. Skipping download.")
+      cli_inform("{.cls {class(self)[[1]]}} FER-2013 already exists. Skipping download.")
```
**todo** idem

cregouby commented on R/dataset-fer.R:
```diff
@@ -107,21 +103,22 @@ fer_dataset <- dataset(
     fs::dir_create(dest_dir)
 
-    cli::cli_inform("Downloading FER-2013 dataset... (Size: ~100MB)")
+    cli_inform("{.cls {class(self)[[1]]}} Downloading FER-2013 dataset... (Size: ~100MB)")
```
**todo** idem

cregouby commented on R/dataset-fgvc.R:
```diff
@@ -79,8 +77,8 @@ fgvc_aircraft_dataset <- dataset(
-    
-    rlang::inform("FGVC-Aircraft dataset (Size: ~2.6 GB) will be downloaded and processed if not already available.")
+
+    cli_inform("{.cls {class(self)[[1]]}} FGVC-Aircraft dataset (Size: ~2.6 GB) will be downloaded and processed if not already available.")
```
**todo** can we switch to the common messages (removing the specifics) ? (apply multiple times)

cregouby commented on R/dataset-fgvc.R:
```diff
@@ -126,7 +124,7 @@ fgvc_aircraft_dataset <- dataset(
     self$labels_df <- merged_df[, levels]
 
-    rlang::inform(glue::glue(
+    cli_inform(glue::glue(
```
**improvement** you can remove the `glue::glue(` that is already included in cli_inform.

cregouby commented on R/dataset-fgvc.R:
```diff
@@ -126,7 +124,7 @@ fgvc_aircraft_dataset <- dataset(
 
-    rlang::inform(glue::glue(
+    cli_inform(glue::glue(
       "FGVC-Aircraft dataset loaded successfully with {length(self$image_paths)} samples ({split}, {annotation_level}-level)."
```
**todo** please generalize this message as well

cregouby commented on R/dataset-flowers.R:
```diff
@@ -89,8 +87,10 @@ flowers102_dataset <- dataset(
     self$target_transform <- target_transform
     self$classes <- self$classes
 
+    cli_inform("{.cls {class(self)[[1]]}} Oxford Flowers 102 dataset will be downloaded and processed if not already cached.")
```
**todo** please remove the specifics in the message and reuse common messages (apply twice)

cregouby commented on R/dataset-mnist.R:
```diff
@@ -247,7 +246,9 @@ qmnist_dataset <- dataset(
     }
 
-    rlang::inform("Processing...")
+    cli_inform("{.cls {class(self)[[1]]}} Downloading...")
+    cli_inform("{.cls {class(self)[[1]]}} Processing...")
```
**todo** those two messages cannot be side by side. Please switch the download message to the common one and move it up to line 241

cregouby commented on R/dataset-mnist.R:
```diff
@@ -447,7 +438,7 @@ emnist_dataset <- dataset(
   initialize = function(root = tempdir(), split = "balanced", transform = NULL, target_transform = NULL,
                         download = FALSE) {
-    rlang::inform(glue::glue(
+    cli_inform(glue::glue(
```
**todo** please switch to the common message (seems like a strong change, but this is fine)

cregouby commented on R/tiny-imagenet-dataset.R:
```diff
@@ -40,12 +40,12 @@ tiny_imagenet_dataset <- torch::dataset(
     fs::file_copy(p, raw_path)
 
-    rlang::inform("Download complete. Now unzipping.")
+    cli_inform("{.cls {class(self)[[1]]}} Download complete. Now unzipping.")
```
**todo** please switch to the common message.

cregouby commented on R/dataset-mnist.R:
**suggestion** merge all mnist variant in a single @rdname mnist.

cregouby commented on R/dataset-flowers.R:
```diff
@@ -40,9 +40,7 @@
-#' @name flowers102_dataset
-#' @aliases flowers102_dataset
-#' @title Oxford Flowers 102 Dataset
+#' @family datasets
```
```suggestion
#' @family classification_dataset
```

cregouby commented on R/dataset-coco.R:
```diff
@@ -55,6 +46,7 @@
 #' masked <- draw_segmentation_masks(item)
 #' tensor_image_browse(masked)
 #' }
+#' @family datasets
```
```suggestion
#' @family detection_dataset
```

cregouby commented on R/tiny-imagenet-dataset.R:
```diff
@@ -8,7 +8,7 @@
 #' @param ... other arguments passed to [image_folder_dataset()].
 #'
-#' @family dataset
+#' @family datasets
```
```suggestion
#' @family classification_dataset
```

cregouby reviewed (commented):
**todo** few missed @family transformation

cregouby commented on R/dataset-flowers.R:
```diff
@@ -89,8 +87,10 @@ flowers102_dataset <- dataset(
     self$target_transform <- target_transform
     self$classes <- self$classes
 
+    cli_inform("{.cls {class(self)[[1]]}} Dataset will be downloaded and processed if not already cached.")
```
**todo** this is a duplicate of L93. Please remove

koshtiakanksha commented:
@cregouby I’ve assigned the datasets to the correct @family now, and I’ve redone the MNIST documentation to address all your suggestions and ensure clean, aggregated output.

Let me know if you prefer this current unified structure, or if I should revert to the previous approach and just apply the suggested changes on top of that. Happy to adjust either way!
